serde_codegen_internals/src/ctxt.rs

```diff
@@ -0,0 +1,43 @@
+use std::fmt::Display;
+use std::cell::RefCell;
+
+#[derive(Default)]
+pub struct Ctxt {
```

Comment:
I'm not very happy with the `Ctxt` structure and it's usage. I'd much rather have regular `Result` type error reporting, but we can leave that to a future refactoring. Maybe the code gets horribly cluttered with `Result`